### PR TITLE
[mono][wasm] Add changes required by emscripten 3.1.4.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/wasm/wasm.proj [hunk 1]
```diff
@@ -65,43 +65,6 @@
   </Target>
 
   <Target Name="GenerateEmccPropsAndRspFiles">
-    <ItemGroup>
```
Comment (vargaz) [inline src/mono/wasm/wasm.proj:68]: These are just moved after the emscripten version checks.

**Comment by vargaz** [comment]
With these changes, the code will work with both emscripten 2.x and 3.x.

File: src/mono/wasm/wasm.proj [hunk 2]
```diff
@@ -123,7 +86,12 @@
       <_EmccVersionRegexPattern>^ *emcc \([^\)]+\) *([0-9\.]+).*\(([^\)]+)\)$</_EmccVersionRegexPattern>
       <_EmccVersion>$([System.Text.RegularExpressions.Regex]::Match($(_EmccVersionRaw), $(_EmccVersionRegexPattern)).Groups[1].Value)</_EmccVersion>
       <_EmccVersionHash>$([System.Text.RegularExpressions.Regex]::Match($(_EmccVersionRaw), $(_EmccVersionRegexPattern)).Groups[2].Value)</_EmccVersionHash>
+	</PropertyGroup>
 
+    <Error Text="Failed to parse emcc version, and hash from the full version string: '$(_EmccVersionRaw)'"
+           Condition="'$(_EmccVersion)' == '' or '$(_EmccVersionHash)' == ''" />
+
+	<PropertyGroup>
```
Comment (radical) [inline src/mono/wasm/wasm.proj:94]:
```suggestion
    </PropertyGroup>

    <Error Text="Failed to parse emcc version, and hash from the full version string: '$(_EmccVersionRaw)'"
           Condition="'$(_EmccVersion)' == '' or '$(_EmccVersionHash)' == ''" />

    <PropertyGroup>
```

File: src/mono/wasm/wasm.proj [hunk 3]
```diff
@@ -139,13 +107,49 @@
...
                       WriteOnlyWhenDifferent="true" />
+
+    <ItemGroup>
+      <_EmccLinkFlags Include="-s EXPORT_ES6=1" Condition="'$(WasmEnableES6)' == 'true'" />
+      <_EmccLinkFlags Include="-s ALLOW_MEMORY_GROWTH=1" />
+      <_EmccLinkFlags Include="-s NO_EXIT_RUNTIME=1" />
+      <_EmccLinkFlags Include="-s FORCE_FILESYSTEM=1" />
+      <_EmccLinkFlags Include="-s EXPORTED_RUNTIME_METHODS=&quot;['FS','print','ccall','cwrap','setValue','getValue','UTF8ToString','UTF8ArrayToString','FS_createPath','FS_createDataFile','removeRunDependency','addRunDependency', 'FS_readFile']&quot;"  />
+      <!-- _htons,_ntohs,__get_daylight,__get_timezone,__get_tzname are exported temporarily, until the issue is fixed in emscripten, https://github.com/dotnet/runtime/issues/64724  -->
+	  <_EmccLinkFlags Include="-s EXPORTED_FUNCTIONS=_free,_malloc,_memalign,_memset" Condition="$([MSBuild]::VersionGreaterThan('$(_EmccVersion)', '3.0'))" />
```
Comment (radical) [inline src/mono/wasm/wasm.proj:122]:
```suggestion
      <_EmccLinkFlags Include="-s EXPORTED_FUNCTIONS=_free,_malloc,_memalign,_memset" Condition="$([MSBuild]::VersionGreaterThan('$(_EmccVersion)', '3.0'))" />
```

**Review by radical — APPROVED** [review]
As long as it builds, and tests pass, LGTM 👍

**Pull request merged**